app/Http/Controllers/ConnectionController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\User;
+use App\Connection;
+use Illuminate\Support\Facades\Auth;
+
+class ConnectionController extends Controller
+{
+    public function show($id)
+    {
+        $user = User::findOrFail($id);
+        $connections = Connection::getAvailableConnections($user);
+
+        return view('users.connections')
+            ->with('user', $user)
+            ->with('connections', $connections);
+    }
+
+    public function add($id)
+    {
+        $connection = new Connection();
+
+        $connection->owner_id = Auth::user()->id;
```

Comment:
The controller shouldn't really know about this. Classes should encapsulate both their behaviour and their data.

Why should this controller know what an "owner_id" and a "user_id" is? Why should it know that they're called that?

What if there was a function like `Connection::connect(User $from, User $to)`.

Wouldn't that be more expressive and encapsulate behaviour better?

Reply:
Yep, it would! ☑️